2-Webpack/main.js

```diff
@@ -0,0 +1,180 @@
+const { v4: uuidv4 } = require("uuid");
+import { LocalStorage } from "./utils/localstorage";
+
+(function () {
+  const LS = new LocalStorage();
+  const editModeItems = [];
+  const todoList = document.getElementById("todoList");
+  const inputText = document.getElementById("addTodo");
+
+  const loadToDoList = (list) => {
+    todoList.innerHTML = "";
+    list.forEach((item) => todoList.appendChild(createItem(list, item)));
+  };
+
+  const addTodo = () => {
+    const newTodo = {
+      text: inputText.value,
+      isFinished: false,
+      id: uuidv4(),
+    };
+    LS.addTodo(newTodo);
+    todoList.insertBefore(
+      createItem(LS.getList(), newTodo),
+      todoList.firstChild
+    );
+    inputText.value = "";
+  };
+
+  const removeTodo = (list, item) => {
+    document.getElementById(item.id).remove();
+    LS.removeTodo(item.id);
+  };
+
+  function removeElement(array, elem) {
+    const index = array.findIndex((el) => el.id == elem.id);
+    if (index > -1) array.splice(index, 1);
+  }
+
+  const renderComponent = (elem, curr, modifiedElem) => {
+    elem.firstChild.replaceChild(curr, modifiedElem);
+  };
+
+  const createTodoItem = (item, todoListItem) => {
+    const todoItem = document.createElement("span");
+    todoItem.innerText = document.getElementById(`${item.id}-editTodo`).value;
+    todoItem.classList.add("todo-text");
+    if (item.isFinished) {
+      todoItem.classList.add("finished-todo");
+      todoListItem.firstChild.checked = true;
+    }
+    return todoItem;
+  };
+
+  const createEditTodoElem = (elemToChange, item) => {
+    const inputText = document.createElement("input");
+    inputText.type = "text";
+    inputText.value = elemToChange.innerText;
+    inputText.classList.add("todo-text", "edit-todo");
+    inputText.id = `${item.id}-editTodo`;
+    return inputText;
+  };
+
+  const editTodo = (item) => {
+    const todoListItem = document.getElementById(item.id);
+    const isFound = editModeItems.some((element) => element.id === item.id);
+    const elemToChange = todoListItem.firstChild.children[1];
```

Comment:
this code is fragile, didn't we agree to rely on `id`s or html attributes?

Reply:
Yes, you right... I fixed it now with querySelector.